go/teleportr/drivers/disburser/driver.go

```diff
@@ -0,0 +1,518 @@
+	"github.com/ethereum-optimism/optimism/go/bss-core/txmgr"
+	"github.com/ethereum-optimism/optimism/go/teleportr/bindings/deposit"
+	"github.com/ethereum-optimism/optimism/go/teleportr/bindings/disburse"
+	"github.com/ethereum-optimism/optimism/go/teleportr/db"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// stateRootSize is the size in bytes of a state root.
+const stateRootSize = 32
+
+var bigOne = new(big.Int).SetUint64(1) //nolint:unused
+
+type Config struct {
+	Name                 string
+	L1Client             *ethclient.Client
+	L2Client             *ethclient.Client
+	Database             *db.Database
+	MaxTxSize            uint64
+	NumConfirmations     uint64
+	DeployBlockNumber    uint64
+	FilterQueryMaxBlocks uint64
+	DepositAddr          common.Address
+	DisburserAddr        common.Address
+	ChainID              *big.Int
+	PrivKey              *ecdsa.PrivateKey
+}
+
+type Driver struct {
+	cfg                  Config
+	depositContract      *deposit.TeleportrDeposit
+	disburserContract    *disburse.TeleportrDisburser
+	rawDisburserContract *bind.BoundContract
+	walletAddr           common.Address
+	metrics              *metrics.Metrics
+
+	currentDepositIDs []uint64
+}
+
+func NewDriver(cfg Config) (*Driver, error) {
+	if cfg.NumConfirmations == 0 {
+		panic("NumConfirmations cannot be zero")
+	}
+	if cfg.FilterQueryMaxBlocks == 0 {
+		panic("FilterQueryMaxBlocks cannot be zero")
+	}
+
+	depositContract, err := deposit.NewTeleportrDeposit(
+		cfg.DepositAddr, cfg.L1Client,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	disburserContract, err := disburse.NewTeleportrDisburser(
+		cfg.DisburserAddr, cfg.L2Client,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	parsed, err := abi.JSON(strings.NewReader(
+		disburse.TeleportrDisburserMetaData.ABI,
+	))
+	if err != nil {
+		return nil, err
+	}
+
+	rawDisburserContract := bind.NewBoundContract(
+		cfg.DisburserAddr, parsed, cfg.L2Client, cfg.L2Client, cfg.L2Client,
+	)
+
+	walletAddr := crypto.PubkeyToAddress(cfg.PrivKey.PublicKey)
+
+	return &Driver{
+		cfg:                  cfg,
+		depositContract:      depositContract,
+		disburserContract:    disburserContract,
+		rawDisburserContract: rawDisburserContract,
+		walletAddr:           walletAddr,
+		metrics:              metrics.NewMetrics(cfg.Name),
+	}, nil
+}
+
+// Name is an identifier used to prefix logs for a particular service.
+func (d *Driver) Name() string {
+	return d.cfg.Name
+}
+
+// WalletAddr is the wallet address used to pay for batch transaction fees.
+func (d *Driver) WalletAddr() common.Address {
+	return d.walletAddr
+}
+
+// Metrics returns the subservice telemetry object.
+func (d *Driver) Metrics() *metrics.Metrics {
+	return d.metrics
```

Comment:
Does this assume that the metrics object is identical to the one in `bss-core`? If so, we should extend this to return an interface of some kind so we can add additional metrics without updating core.